src/Compilers/Core/Portable/ReferenceManager/AssemblyReferenceBinding.cs

```diff
@@ -13,9 +15,9 @@ internal partial class CommonReferenceManager<TCompilation, TAssemblySymbol>
         /// Result of binding an AssemblyRef to an AssemblyDef. 
         /// </summary>
         [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
-        internal struct AssemblyReferenceBinding
+        internal readonly struct AssemblyReferenceBinding
         {
-            private readonly AssemblyIdentity _referenceIdentity;
+            private readonly AssemblyIdentity? _referenceIdentity;
```

Comment:
Both constructors use non-nullable `referenceIdentity`. I assume this field is marked as nullable because one might do `default(AssemblyReferenceBinding)`. Is that the reason?

Reply:
Yes. The runtime nullable annotation guide favors treating reference fields in structs as nullable since the use case `default(T)` must be considered. There are some exceptions considered but my reading is that it should be the rare case. 

https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/api-guidelines/nullability.md